File: Runtime/src/Core/CoreScript/BCL/IEnumerator.js
///////////////////////////////////////////////////////////////////////////////
// IEnumerator

ss.IEnumerator = function#? DEBUG IEnumerator$##() { };
#if DEBUG
ss.IEnumerator.prototype = {
    get_current: null,
    moveNext: null,
    reset: null
}
#endif // DEBUG

ss.IEnumerator.registerInterface('ss.IEnumerator', ss.IDisposable);

File: Runtime/CoreLib/Collections/Generic/IEnumerator.cs
// IEnumerator.cs
// Script#/Libraries/CoreLib
// This source code is subject to terms and conditions of the Apache License, Version 2.0.
//

using System.Runtime.CompilerServices;

namespace System.Collections.Generic {
	[IncludeGenericArguments(false)]
	[ScriptNamespace("ss")]
	[ScriptName("IEnumerator")]
	[Imported(ObeysTypeSystem = true)]
	public interface IEnumerator<out T> : IDisposable, IEnumerator {
		new T Current { [ScriptName("current")] get; }
		new bool MoveNext();
		new void Reset();
	}
}
